apollo-annotations/src/appleMain/kotlin/com/apollographql/apollo3/annotations/ApolloAdaptableWith.kt

```diff
@@ -0,0 +1,10 @@
+package com.apollographql.apollo3.annotations
+
+import kotlin.reflect.AssociatedObjectKey
+import kotlin.reflect.ExperimentalAssociatedObjects
+import kotlin.reflect.KClass
+
+@OptIn(ExperimentalAssociatedObjects::class)
+@AssociatedObjectKey
```

Comment:
TIL! Very cool feature! 👍

Reply:
Not sure I want to know how it's implemented 😅 ...
... but it's useful!